File: src/models/StoryRating.ts
import {Entity,Column,ManyToOne,JoinColumn,PrimaryColumn,} from 'typeorm';
import { User } from './User';
import { Story } from './Story';

@Entity('story_rating')
export class StoryRating {
  @PrimaryColumn()
  user_id!: number;

  @PrimaryColumn()
  story_id!: number;

  @Column({ type: 'int' })
  value!: number;

  @Column({ type: 'varchar', nullable: true })
  description!: string;

  @ManyToOne(() => User, (user) => user.ratings)
  @JoinColumn({ name: 'user_id' })
  user!: User;

  @ManyToOne(() => Story, (story) => story.ratings)
  @JoinColumn({ name: 'story_id' })
  story!: Story;
}

File: src/models/story_rating.py
from sqlalchemy import Column, Integer, String, ForeignKey
from sqlalchemy.orm import relationship
from src.db.database import Base


class StoryRating(Base):
    __tablename__ = "story_rating"

    user_id = Column(Integer, ForeignKey("users.id"), primary_key=True)
    story_id = Column(Integer, ForeignKey("story.id"), primary_key=True)
    value = Column(Integer, nullable=False)
    description = Column(String, nullable=True)

    user = relationship("User", back_populates="ratings")
    story = relationship("Story", back_populates="ratings")
